### PR TITLE
Dponomarenko/get addressed by postcode

### PR DESCRIPTION
**Pull request by Sameperson**

### CODE REVIEW

File: src/main/java/com/opinta/controller/DictionariesController.java [hunk 1]
```diff
@@ -67,4 +73,10 @@ public DictionariesController(TariffGridService tariffGridService,
             return new ResponseEntity<>(NOT_FOUND);
         }
     }
+
+    @RequestMapping("/addresses")
+    public ResponseEntity<?> getAddressesByPostcode(@RequestParam(value="postcode") String postcode) {
+        List<CityDto> cities = cityService.getAllCitiesByPostcode(postcode);
```
Comment (Diarsid) [inline src/main/java/com/opinta/controller/DictionariesController.java]: Please, change 
@RequestParam(value="postcode") String postcode
to
@RequestParam String postcode

The reason is if annotated argument name matches with real http request parameter name, Spring will discover and inject it even without explicit (value = "...") description.

File: src/main/resources/db/migration/dev/V3.4__populate_city.sql [hunk 2]
```diff
@@ -7,6 +7,7 @@ INSERT INTO City (id, country_id, region_id, district_id, name) VALUES ('35385',
 INSERT INTO City (id, country_id, region_id, district_id, name) VALUES ('35386','UA','2','39','Мощена');
 INSERT INTO City (id, country_id, region_id, district_id, name) VALUES ('35387','UA','2','39','Новий Мосир');
 INSERT INTO City (id, country_id, region_id, district_id, name) VALUES ('35388','UA','2','39','Нові Кошари');
+INSERT INTO City (id, country_id, region_id, district_id, name) VALUES ('42309','UA','10','185','Київ');
```
Comment (loveforjava) [inline src/main/resources/db/migration/dev/V3.4__populate_city.sql:10]: The main rule is that you can't change the script that has already been applied. You should add new version instead.
It's ok for now, cuz this script hasn't been applied yet in prod.

File: src/main/java/com/opinta/service/DictionaryService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.opinta.service;
+
+import com.opinta.entity.classifier.City;
+
+import java.util.List;
+
+public interface DictionaryService {
+
+    List<City> getAllCitiesByPostcode(String postcode);
+
```
Comment (loveforjava) [inline src/main/java/com/opinta/service/DictionaryService.java]: Plz, delete this empty line

File: src/main/java/com/opinta/service/CityService.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.opinta.service;
+
+import com.opinta.dto.CityDto;
+import java.util.List;
+
+public interface CityService {
+
+    List<CityDto> getAllCitiesByPostcode(String postcode);
+
```
Comment (loveforjava) [inline src/main/java/com/opinta/service/CityService.java]: Plz, delete empty line

File: src/main/java/com/opinta/config/HibernateConfig.java [hunk 5]
```diff
@@ -34,6 +34,22 @@
...
+    @Value("classpath:db/migration/prod/V3.3__populate_district.sql")
+    private Resource districtPopulatorProd;
+    @Value("classpath:db/migration/prod/V3.4__populate_city.sql")
+    private Resource cityPopulatorProd;
+    @Value("classpath:db/migration/prod/V3.5__populate_countryside_postcode.sql")
+    private Resource countrysidePostcodePopulatorProd;
+    @Value("classpath:db/migration/prod/V3.6__populate_tariff_grid.sql")
+    private Resource tariffGridPopulatorProd;
+    @Value("classpath:db/migration/prod/V3.7_populate_city_postcode.sql")
+    private Resource postcodePopulatorProd;
```
Comment (loveforjava) [inline src/main/java/com/opinta/config/HibernateConfig.java]: Why do you need all this variables here for prod?

File: src/main/java/com/opinta/config/HibernateConfig.java [hunk 6]
```diff
@@ -206,6 +224,23 @@ public DatabasePopulator databasePopulatorProduction() {
         populator.addScript(cityPopulatorMemory);
         populator.addScript(countrysidePostcodePopulatorMemory);
         populator.addScript(tariffGridPopulatorMemory);
+        populator.addScript(postcodePopulatorMemory);
         return populator;
     }
+
+    @Bean(name = "databasePopulator")
+    @Profile("prod")
+    public DatabasePopulator databasePopulatorProduction() {
```
Comment (loveforjava) [inline src/main/java/com/opinta/config/HibernateConfig.java]: Why do you need separate data populator for Oracle db?

File: src/main/java/com/opinta/controller/DictionariesController.java [hunk 7]
```diff
@@ -66,4 +72,10 @@ public DictionariesController(TariffGridService tariffGridService,
             return new ResponseEntity<>(NOT_FOUND);
         }
     }
+
+    @RequestMapping("/addresses")
+    public ResponseEntity<?> getAddressesByPostcode(@RequestParam String postcode) {
+        List<CityDto> cities = cityService.getAllCitiesByPostcode(postcode);
+        return new ResponseEntity<>(cities, OK);
```
Comment (loveforjava) [inline src/main/java/com/opinta/controller/DictionariesController.java]: Plz, do it in one line.

File: src/main/java/com/opinta/dao/CityDao.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.opinta.entity.classifier.City;
+import com.opinta.entity.classifier.CityPostcode;
+
+import java.util.List;
+
+public interface CityDao {
+
+    List<City> getAllCitiesByPostcode(String postcode);
+
```
Comment (loveforjava) [inline src/main/java/com/opinta/dao/CityDao.java]: Plz, delete empty line

File: src/main/resources/sql/dev/db-data-city-postcode.sql [hunk 9]
```diff
@@ -0,0 +1,25 @@
+DELETE FROM CityPostcode ;
```
Comment (loveforjava) [inline src/main/resources/sql/dev/db-data-city-postcode.sql]: The folder sql in the resources is redundant. With flyway we don't need it any more

File: src/main/java/com/opinta/service/DictionaryService.java (hunk 3 above)
Comment (Sameperson) [reply src/main/java/com/opinta/service/DictionaryService.java]: It´s interface, we have empty line literally everywhere in the interfaces

**Review by loveforjava — APPROVED** [review]

**Pull request merged**